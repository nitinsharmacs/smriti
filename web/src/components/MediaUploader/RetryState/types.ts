import type { MediaItemType } from 'src/components/MediaUploader/types';

export interface RetryStateProps {
  achievedUploads: number;
  targetUploads: number;
  mediaItems: MediaItemType[];
  onRetry?: () => void;
}
