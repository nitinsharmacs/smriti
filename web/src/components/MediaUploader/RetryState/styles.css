.retry-modal-heading {
  margin-bottom: 1em;
}
