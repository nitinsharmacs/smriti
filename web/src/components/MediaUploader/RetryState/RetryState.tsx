import Button from 'src/components/Button/Button';
import Modal from 'src/components/MediaUploader/Modal/Modal';
import ExpandableModal from 'src/components/MediaUploader/ExpandableModal/ExpandableModal';
import MediaItem from 'src/components/MediaUploader/MediaItem/MediaItem';
import type { RetryStateProps } from 'src/components/MediaUploader/RetryState/types';

import './styles.css';

const RetryState = (props: RetryStateProps) => {
  return (
    <Modal>
      <ExpandableModal
        expandableContent={props.mediaItems.map((item) => (
          <MediaItem
            key={item.id}
            name={item.name}
            type={item.type}
            status={item.status}
            progress={item.progress}
          />
        ))}
        actions={[
          <Button key='retry-btn' variant='contained' onClick={props.onRetry}>
            Retry
          </Button>,
        ]}
      >
        <h4 className='retry-modal-heading'>
          {props.targetUploads - props.achievedUploads} out of{' '}
          {props.targetUploads} uploads remaining
        </h4>
      </ExpandableModal>
    </Modal>
  );
};

export default RetryState;
