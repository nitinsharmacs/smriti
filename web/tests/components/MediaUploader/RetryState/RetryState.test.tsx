import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import RetryState from 'src/components/MediaUploader/RetryState/RetryState';
import {
  MediaType,
  MediaUploadStatus,
} from 'src/components/MediaUploader/types';
import { describe, expect, it, vi } from 'vitest';

describe('RetryState', () => {
  it('should render retry state', () => {
    const { container } = render(
      <RetryState achievedUploads={0} targetUploads={0} mediaItems={[]} />
    );
    expect(container).toMatchSnapshot();
  });

  it('should show more/less', async () => {
    render(
      <RetryState
        achievedUploads={0}
        targetUploads={2}
        mediaItems={[
          {
            id: '123',
            type: MediaType.Image,
            status: MediaUploadStatus.Failed,
            progress: 12,
            name: 'image.png',
          },
          {
            id: '1234',
            type: MediaType.Video,
            status: MediaUploadStatus.Failed,
            progress: 40,
            name: 'video.mp4',
          },
        ]}
      />
    );

    const showMoreBtn = screen.getAllByRole('button')[1];

    expect(screen.queryByText('image.png')).not.toBeInTheDocument();
    expect(screen.queryByText('video.mp4')).not.toBeInTheDocument();

    expect(showMoreBtn.textContent).toBe('Show more');

    await userEvent.click(showMoreBtn);

    expect(showMoreBtn.textContent).toBe('Show less');
    expect(screen.getByText('image.png')).toBeInTheDocument();
    expect(screen.getByText('video.mp4')).toBeInTheDocument();
  });

  it('should retry upload', async () => {
    const retryMock = vi.fn();

    render(
      <RetryState
        achievedUploads={0}
        targetUploads={1}
        mediaItems={[
          {
            id: '123',
            type: MediaType.Image,
            status: MediaUploadStatus.InProgress,
            progress: 12,
            name: 'image.png',
          },
        ]}
        onRetry={retryMock}
      />
    );

    const retryBtn = screen.getAllByRole('button')[0];

    await userEvent.click(retryBtn);

    expect(retryMock).toHaveBeenCalledOnce();
  });
});
