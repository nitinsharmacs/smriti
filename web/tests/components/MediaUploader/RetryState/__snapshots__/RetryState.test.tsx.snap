// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RetryState > should render retry state 1`] = `
<div>
  <div
    class="upload-txn-modal"
  >
    <div
      class="expandable-modal"
    >
      <div
        class="content"
      >
        <h4
          class="retry-modal-heading"
        >
          0
           out of
           
          0
           uploads remaining
        </h4>
      </div>
      <div
        class="actions"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary contained-btn css-74d805-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Retry
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary text-btn css-1uent87-MuiButtonBase-root-MuiButton-root"
          style="margin-left: 0.5em;"
          tabindex="0"
          type="button"
        >
          Show more
        </button>
      </div>
      <div
        class="expandable-content"
      >
        <div
          class="collapsable-media-items"
        />
      </div>
    </div>
  </div>
</div>
`;
